backend/src/controller/donors.controllers.js

```diff
@@ -0,0 +1,156 @@
+const donorService = require("../services/donor.service.js");
+const Joi = require("joi");
+
+async function addDonor(req, res) {
+    // Joi validation
+    const schema = Joi.object().keys({
```

Comment:
needs to implement a separate middleware

Reply:
added validation.js to middleware,  from that all validation function are exported and will be used as middleware in api